\documentclass[12pt]{article}

\usepackage{parskip}
\setlength{\parindent}{0pt}
\setlength{\parskip}{1.3ex plus 0.5ex minus 0.3ex}

\usepackage{amsmath}
\usepackage{graphicx}
\usepackage{placeins}
\usepackage{color}
\usepackage{hyperref}
\hypersetup{
    colorlinks,
    citecolor=blue,
    filecolor=blue,
    linkcolor=blue,
    urlcolor=blue
}
\graphicspath{{./images/}}

\title{SRA analysys}
\author{Zebulun Arendsee}
\date{\today}
\begin{document}

\maketitle

\section{Introduction}
\label{sec:introduction}

\section{Methods}
\label{sec:methods}

\subsection{Kallisto}
\label{sub:kallisto}

Kallisto is a very fast tool for quantitative analysis of transcriptome
expression \cite{bray_near-optimal_2015}.

\subsection{RNA-seq cleaning and quality control}
\label{sub:rna_seq_cleaning}

Raw RNA-seq data should be checked for sequence quality, GC content, and
over-represeneted k-mers and duplicate reads. This can be done with NGSQC
\cite{patel_ngs_2012}. 

\begin{figure}[!hbpt] \centering
  \includegraphics[width=0.9\textwidth]{ngsqc-flowchart}
  \caption{Tools used in NGSQC}
\end{figure}
\FloatBarrier


\subsection{General RNA-seq notes}
\label{sub:general_rna_seq_notes}

Here is a nice review \cite{conesa_survey_2016}.

\begin{figure}[!hbpt] \centering
  \includegraphics[width=0.9\textwidth]{illumina-fig4}
  \caption{Paired-end reads. From \href{http://www.illumina.com/technology/next-generation-sequencing/paired-end-sequencing\_assay.html}{Illumina website}}.
\end{figure}

\subsection{Short Read Archive id}

\begin{description}
  \item[SRP] study id
    \begin{itemize}
      \item a study is a project with an overall goal
      \item is a collection of experiments 
    \end{itemize}
  \item[SRX] experiment id
      \begin{itemize}
        \item the output set of an instrument
        \item may include many samples
      \end{itemize}
  \item[SRS] sample id
      \begin{itemize}
        \item a sample refers to a biological set of entities
        \item basically, one eppendorf tube
        \item a sample may contain many runs
      \end{itemize} 
  \item[SRR] run id
      \begin{itemize}
        \item details on data from a single run on a single sample
      \end{itemize}
  \item[SRA] short read accession id
      \begin{itemize}
        \item Each submission to SRA has an associated accession.
        \item The SRA is not always monophyletic
      \end{itemize}
\end{description}

The 'S' can be replaced with 'E' or 'D', e.g. SRA, ERA, DRA. 'S' indicates the
entry is from NCBI, E from EBI, and D from DDBJ.

\section{Resources}
\label{sec:resources}

The memory required in fastq is about equal to 4 times the number of bases.

In Arabidopsis, there are 30T bases, or 120 Tb of fastq data.

In yeast, there are 4.7

4050 97082164504 8837 111 043 572

50B spots, 4T bases
4054 48973262859 3,715,407,919,425

The largest number of bases per entry in Arabidopsis is 50.

5.9gb fastq against yeast in 22s

\FloatBarrier
\bibliographystyle{plain}
\bibliography{report}
\end{document}
